pkg/mutation/schema/node.go

```diff
@@ -128,7 +150,27 @@ func (n *node) conflicts(childKey string) IDSet {
 		nTypes--
 	}
 
+	// Check if we need to terminate in this child node
+	mustTerminate := false
+	for _, child := range n.Children[childKey] {
+		if len(child.MustTerminate) > 0 {
+			mustTerminate = true
+			break
+		}
+	}
+
 	for nodeType, child := range n.Children[childKey] {
+		// found a conflict if we need to terminate at this child node and there
+		// are different types of children with this key.
+		if mustTerminate && len(n.Children[childKey]) > 1 {
```

Comment:
This doesn't seem quite right... this looks like it's testing for siblings of a node that must terminate?

I think the salient thing here is... if there is a node that must terminate, it should have no children. Any mutators that reference a child of a node where `MustTerminate` is populated are conflicting with the members of the `MustTerminate` set.

I think you are right to aggregate `mustTerminate` across the children above (lines 154-160), but below we need to check that the aggregate # of children for all nodes with the same child key is zero.